### PR TITLE
Fix ConvTranspose: enhance attribute check

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**

- Enhance the attribute check for ConvTranspose: `pads` and `auto_pads` can not be used at the same time.
- Add a unit test for it
- (move it to another PR) Update the document for ConvTranspos: make the example be consistent as the doc description

**Motivation and Context**
https://github.com/onnx/onnx/issues/2787

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -10743,7 +10743,7 @@ This version of the operator has been available since version 11 of the default
   output_shape can also be explicitly specified in which case pads values are auto generated using these equations:
   
     total_padding[i] = stride[i] * (input_size[i] - 1) + output_padding[i] + ((kernel_shape[i] - 1) * dilations[i] + 1) - output_shape[i]
-    If (auto_pads != SAME_UPPER): pads[start_i] = total_padding[i]/2; pads[end_i] = total_padding[i] - (total_padding[i]/2)
+    If (auto_pads == SAME_UPPER): pads[start_i] = total_padding[i]/2; pads[end_i] = total_padding[i] - (total_padding[i]/2)
```
Comment (askhade) [inline docs/Changelog.md:10746]: Is this intentional? Why change this ?

File: onnx/defs/nn/defs.cc [hunk 2]
```diff
@@ -1300,7 +1304,7 @@ If the pads parameter is provided the shape of the output is calculated via the
 output_shape can also be explicitly specified in which case pads values are auto generated using these equations:
 
   total_padding[i] = stride[i] * (input_size[i] - 1) + output_padding[i] + ((kernel_shape[i] - 1) * dilations[i] + 1) - output_shape[i]
-  If (auto_pads != SAME_UPPER): pads[start_i] = total_padding[i]/2; pads[end_i] = total_padding[i] - (total_padding[i]/2)
+  If (auto_pads == SAME_UPPER): pads[start_i] = total_padding[i]/2; pads[end_i] = total_padding[i] - (total_padding[i]/2)
```
Comment (askhade) [inline onnx/defs/nn/defs.cc:1307]: same comment as above - why is this change needed?

File: docs/Changelog.md (hunk 1 above)
Comment (jcwchen) [reply docs/Changelog.md:10746]: It is another old issue related to ConvTranspose which hasn't been resolved: https://github.com/onnx/onnx/issues/1609 There is also a PR for it: https://github.com/onnx/onnx/pull/2097
The original reason is the description is not consistent with the example. I thought it's a good chance to push this forward so I added this fix in this PR as well.

**Comment by jcwchen** [comment]
Updating the document for ConvTranspos will move to another PR because the modification needs further review.

**Comment by askhade** [comment]
@gramalingam @ebarsoum @postrational : This PR enforces a condition which is already part of the spec. I do not think we need a version bump here. Thoughts?

**Review by askhade — APPROVED** [review]

File: onnx/defs/nn/defs.cc [hunk 3]
```diff
@@ -1191,6 +1191,10 @@ void convTransposeShapeInference(InferenceContext& ctx) {
     if (pads.size() != n_input_dims * 2) {
       fail_shape_inference("Attribute pads has incorrect size");
     }
+    const auto* auto_pad_attr = ctx.getAttribute("auto_pad");
+    if (nullptr != auto_pad_attr) {
+      fail_shape_inference("The pads attribute cannot be used simultaneously with auto_pad attribute");
```
Comment (gramalingam) [inline onnx/defs/nn/defs.cc:1196]: Just a minor nit: this is neither a shape_inference error nor a type_inference error. Unfortunately, we have only "fail_type_inference" and "fail_shape_inference". This is okay, but it might be good to have a consistent categorization of errors (may be adding another category of errors) ... not necessarily in this PR, could be later also.

**Review by gramalingam — APPROVED** [review]

**Comment by gramalingam** [comment]
> This PR enforces a condition which is already part of the spec. I do not think we need a version bump here. Thoughts?
Agree, this seems fine.

**Comment by gramalingam** [comment]
> @gramalingam @ebarsoum @postrational : This PR enforces a condition which is already part of the spec. I do not think we need a version bump here. Thoughts?

Agree
